### PR TITLE
Escape variable names if needed

### PR DESCRIPTION
**Pull request by angrykoala**

### CODE REVIEW

File: src/utils/escape.ts [hunk 1]
```diff
@@ -34,6 +34,11 @@ export function escapeProperty(propName: string): string {
     return escapeIfNeeded(propName);
 }
 
+/** Escapes a variable name if needed */
+export function escapeVariable(varName: string): string {
+    return escapeIfNeeded(varName);
+}
```
Comment (darrellwarde) [inline src/utils/escape.ts:40]: Do we ever envision a world where escaping for types, labels, properties and variables have different behaviours?

**Review by darrellwarde — APPROVED** [review]
LGTM!

As a general side note, might we want to consider something similar to the TCK test validation we have in the CI for the GraphQL Library to assert the validity of all Cypher generated by the Cypher Builder?

File: src/utils/escape.ts (hunk 1 above)
Comment (angrykoala) [reply src/utils/escape.ts:40]: Well, for a while it was different, hence these separate methods. I can imagine some edge cases we may want to treat differently and I think it makes sense, in terms of separation of concepts, to have separate functions, even if all of those pipe to the same function atm

**Comment by angrykoala** [comment]
That is a good point, although not all of the generated Cypher is valid Cypher (Cypher-Builder does not guarantee valid Cypher and some tests do not generate full queries)

Definitely something to add at some point

**Pull request merged**